File: backend/src/models/ScrapingLog.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const ScrapingLog = sequelize.define('ScrapingLog', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  task_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  url: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  status: {
    type: DataTypes.STRING(20),
    defaultValue: 'PENDING',
  },
  error_message: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  pages_crawled: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  phones_extracted: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  emails_extracted: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'scraping_logs',
  timestamps: false,
});

module.exports = ScrapingLog;

File: backend/app/models/scraping_log.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text, func
from sqlalchemy.orm import relationship
from app.database.database import Base


class ScrapingLog(Base):
    __tablename__ = "scraping_logs"

    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("scraping_tasks.id"), nullable=False)

    url = Column(Text, nullable=True)
    status = Column(String(20), default="SUCCESS")  # SUCCESS/FAILED/BLOCKED/TIMEOUT/SKIPPED
    error_message = Column(Text, nullable=True)
    pages_crawled = Column(Integer, default=0)
    phones_extracted = Column(Integer, default=0)
    emails_extracted = Column(Integer, default=0)
    created_at = Column(DateTime, server_default=func.now())

    task = relationship("ScrapingTask", back_populates="logs")
